HomeWork3

Shamayko opened the pull request:
HomeWork 3

Shamayko committed:
Первая попытка сдать домашнее задание через git.

Shamayko committed:
Домашнее задание 3.

emgusev commented on src/ru/geekbrains/homeworks/homework3/HomeWork3.java:
```diff
@@ -0,0 +1,197 @@
+         * проверки преподавателя (наверное, но это не точно).
+         */
+
+        int[] arr = {1, 0, 0, 1};
```
Такие штуки надо делать отдельными методами. написание кода строчка за строчкой зовется "простыня" :)

Shamayko replied:
Поправил, убрал все простыни, сделал методы. 
Не уверен правильно ли все делаю в гите - вот ссылка на сам коммит на всякий случай.
https://github.com/Shamayko/HomeWorks/commit/f5814f73595c7941d7846ae88d2a54853a18c00d

emgusev commented on src/ru/geekbrains/homeworks/homework3/HomeWork3.java:
```diff
@@ -0,0 +1,197 @@
+                for (int j = 0; j < table[i].length; j++) {
+                    if (table[i] == table[j]) {
+                        System.out.print(1);
+                    } else if (i + j == 4) {
```
Вот тут явная завязка на число. Не надо так) Ты знаешь размер массива.

Shamayko replied:
Понял, был не прав :)  убрал завязку на число, сделал логику.

emgusev commented on src/ru/geekbrains/homeworks/homework3/HomeWork3.java:
```diff
@@ -0,0 +1,197 @@
+ * Задача 4
+ * Вышиваем крестиком )))
+ */
+        int[][] table = new int[5][5];
```
А само заполнение массива я так и не увидел)  то есть если я после цикла выведу массив на экран, то там будут 0 везде

Shamayko replied:
```suggestion
        int[][] table = new int[5][5];
```
Скажите пожалуйста, согласно задания заполнять надо только диагональные элементы, поэтому остальные не заполнял. Чем необходимо заполнить другие элементы? Каким-то любым одним числом или по порядку числами от 1 до 25?

emgusev replied:
А чем они будут заполнены по умолчанию? :) 

Я к тому, что ожидал увидеть arr[i][j] = 1 вместо system.out

В явном виде присвоения не написано.

Shamayko replied:
Понял, у меня был вывод в печать, а не заполнение массива. Заполнил массив, не диагонали заполнил 8ками.

emgusev commented on src/ru/geekbrains/homeworks/homework3/HomeWork3.java:
```diff
@@ -0,0 +1,197 @@
+        int chek = 0;
+        for (int i = 0; i < sarr.length; i++) {
+            chek = chek + sarr[i];
+            if (chek == sum - chek) {
```
[info]Для оптимизации

if(chek > sum/2) {
break;
}
Дальше смотреть смысла уже нет

Shamayko replied:
Спасибо, добавил break.

emgusev commented on src/ru/geekbrains/homeworks/homework3/HomeWork3.java:
```diff
@@ -0,0 +1,197 @@
+
+
+        for (int i = 0; i < n; i++) {
+            for (int j = 0; j < arr.length - 1; j++) {
```
Вот тут работать не будет. Цикличности не хватает. Последний элемент должен стать первым.

Shamayko replied:
Вот тут не понял. Покрутил, проверил - делает последнее число первым и наоборот (в зависимости от отрицательности переменной).

emgusev replied:
Да, ок. Метод пузырька. Сбило с толку, что направления реверсированы. Сдвиг не в ту сторону проверял и картинка не сошлась

Shamayko committed:
Домашнее задание 3.

Shamayko committed:
Домашнее задание 4. Крестики нолики.
Программа для поля 5 на 5, с победой при 4х в ряд. Программа проверяет ниличие у игрока 2х занятых клеток по диагонали (не обязательно в ряд, вдруг мы потом коварно их соединим), приоритетно пытается блокировать их. Если их не видит начинает проверку горизонтали на предмет 3х занятых, затем аналогично вертикали, затем опять горизонтали для 2х, затем вертикали. Если ни одной угрозы не обнаружено ставит нолик в свободную клетку.
